### ISSUE TITLE
[admin] CI has started failing regularly try to create a kind cluster

### ISSUE THREAD

**Issue by TylerHelmuth** [open-telemetry/opentelemetry-operator#1863]
I noticed this week that our CI has started failing consistently for any CI job trying to create a kind cluster.  See https://github.com/open-telemetry/opentelemetry-operator/actions/runs/5347042202/jobs/9695625228?pr=1850 as an example.

**Comment by swiatekm-sumo** [comment]
Looks like this is the result of a kind version upgrade from 1.19 to 1.20 in Github Runners. See the most recent release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230619.1.

The oldest version kind 0.20.0 officially supports is 1.21: https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0, and I don't see a version which supports the same 1.19-1.27 range as we do. Maybe it's time to drop support for 1.19 and 1.20? If we want to keep them, the alternative would be to either use different kind versions or build our own Node images.

**Comment by swiatekm-sumo** [comment]
The tests succeed if we upgrade the kind Node images and drop 1.19 and 1.20: https://github.com/open-telemetry/opentelemetry-operator/pull/1864.

**Comment by TylerHelmuth** [comment]
@swiatekm-sumo based on your investigation how difficult would it be to use a different kind version for the 1.19 and 1.27 targets?

**Comment by pavolloffay** [comment]
From Red Hat support perspective we are fine dropping testing for 1.19 and 1.20. The oldest supported OpenShift version is 4.10 which is based on k8s 1.23.

**Comment by swiatekm-sumo** [comment]
From Sumo perspective, we basically support the oldest version major cloud providers support, which in this case is also 1.23. If the cost of supporting 1.19 and 1.20 is just some additional CI complexity, then maybe it's worth it anyway?

> @swiatekm-sumo based on your investigation how difficult would it be to use a different kind version for the 1.19 and 1.27 targets?

I don't think it's a lot of work, I'll try to get it working tonight. I do think that if we keep 1.19, we should consider *also* running tests on the oldest K8s version supported by upstream - currently 1.24, soon to be 1.25.

**Comment by pavolloffay** [comment]
>  we should consider also running tests on the oldest K8s version supported by upstream - currently 1.24, soon to be 1.25

We discussed this in the past, and decided to run e2e tests only on the oldest and newest versions to save some CI time (assuming the versions in between should work)

**Comment by TylerHelmuth** [comment]
I think bumping the minimum supported version to 1.21 is reasonable if the rest of the industry has.

**Comment by jaronoff97** [comment]
I'm in support of bumping to 1.21. 1.19 and 1.20 have been EOL for [more than a year](https://kubernetes.io/releases/patch-releases/#non-active-branch-history).

**Comment by swiatekm-sumo** [comment]
As using different kind versions is straightforward, my suggestion would be to do that to unblock the CI in the short term. We can also drop support for the K8s versions below 1.22, but that should be done with more fanfare in a separate issue IMO. 

See https://github.com/open-telemetry/opentelemetry-operator/pull/1864.

**Comment by swiatekm-sumo** [comment]
Can we close this? Looks like the CI problem is fixed, and K8s version support should be discussed in its own issue.
